### PR TITLE
task: input style

### PR DESCRIPTION
**Pull request by ManuViola77**
### Pull request type

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

This PR adapts the current Input component to the styles in the Figma designs.

---

#### Jira board reference

- [Input Component](https://rootstrap.atlassian.net/browse/TH-146)

---

#### Notes:

- Just to see the disabled case I set that to the email, but that change is not committed since here it is actually editable.

---

#### Tasks:

- [x] Tested on IOS
- [x] Tested on Android
- [ ] Tested on a small device
- [ ] Tested on a real device
- [ ] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

---

#### Preview

**iOS:**

**Android:**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "@tanstack/react-query": "^4.2.1",
     "axios": "^0.27.2",
     "dripsy": "^3.10.0",
+    "eslint-plugin-prettier": "^4.2.1",
```
Comment (jpirazusta) [inline package.json:51]: Same as already commented in [#20](https://github.com/rootstrap/the-hive-react-native/pull/20): I think we should add this to devDependencies

File: src/design-system/components/Input/index.tsx [hunk 2]
```diff
@@ -56,49 +45,124 @@ const ForwardInputRefFunction: React.ForwardRefRenderFunction<RNTextInput, TextI
...
+    }: DefaultColors = {}) => {
+      const disabledColor = isDisabled ? disabled : undefined;
+      const invalidColor = showError ? errorColor : undefined;
+      const successColor = isSuccess ? success : undefined;
+      const focusColor = isFocused ? focus : undefined;
+      const filledColor = isFilled ? filled : undefined;
+      const currentColor =
+        disabledColor ?? focusColor ?? invalidColor ?? successColor ?? filledColor ?? defaultColor;
+
+      return currentColor;
```
Comment (jpirazusta) [inline src/design-system/components/Input/index.tsx:82]: Maybe is this what it doesn't like to code climate. Maybe we can try something like this?

```suggestion
if (isDisabled) return disabled;
if (showError) return errorColor;
if (isSuccess) return success;
if (isFocused) return focus;
if (isFilled) return filled;
return defaultColor;
```

**Review by jpirazusta — APPROVED** [review]
🙌

File: src/design-system/components/Input/types.ts [hunk 3]
```diff
@@ -29,8 +30,11 @@ export type TextInputProps = {
...
-  isFocused: boolean;
-  isInvalid: boolean;
-  isSuccess?: boolean;
+export type DefaultColors = {
+  disabled?: keyof Colors;
+  error?: keyof Colors;
+  success?: keyof Colors;
+  focus?: keyof Colors;
+  filled?: keyof Colors;
+  default?: keyof Colors;
```
Comment (nicmalegre) [inline src/design-system/components/Input/types.ts:39]: Maybe you could create a type `keyof Colors` to avoid duplicating

**Review by nicmalegre — APPROVED** [review]

**Pull request merged**